### DISCUSSION TITLE
Almost inaudible

### DISCUSSION THREAD

**Discussion by romix123** [marcel-licence/esp32_basic_synth#61] [Q&A] [answered]
I have built and compiled the setup as explained in the various docs.
The test sound is audible in my headphones, but actual keystrokes with my midi keyboards (arturio minilab ii and another one) are barely audible. I can’t find what is wrong. If I connect the pcm board to an amp, I still need to pump up the volume to almost max. This can’t be right.

**Comment by Sicco55** [comment, marked as answer]
Per default the main filter cutoff: 0.000. Give it a higher value.

**Reply by marcel-licence** [reply]
Thx. In addition to that the velocity value may be different from the MIDI controller. You can also check this and set a constant value instead. (Not sure if it is implemented in this project but it may also cause a difference)

**Reply by romix123** [reply]
The main filter cutoff did the trick. I mapped one of the pots on my Arturia to SYNTH_PARAM_MAIN_FILT_CUTOFF and now we're good to go.

**Reply by marcel-licence** [reply]
Do you have the ADC part active in the code? I was thinking about checking the default values. It may happen that with ADC active but nothing connected all controls will modified when starting the firmware.

**Reply by marcel-licence** [reply]
Following could be added at the end of setup():

`Synth_SetParam(SYNTH_PARAM_MAIN_FILT_CUTOFF, 1.0f);`

If that could fix the problem, then I would like to update the code =)

**Comment by romix123** [comment]
No, just the pcm5102 board

Verstuurd vanaf mijn iPad

**Comment by romix123** [comment]
Another thing. My setup misses the occasional note off. Only way to mute is strike the missed tone again.
Apart from that. Love your work.


Verstuurd vanaf mijn iPad

**Reply by marcel-licence** [reply]
Thank you! Hm but missing note offs are very frustrating. After soldering a little circuit for the MIDI input I improved my experience. But I had also 2 missing note offs withing ~2 hours. I do not know how the input will be processed. Hopefully there is something left which can be fixed.

**Reply by MichaelPNolan** [reply]
My USB midi also loses the note off from time to time. My stand-alone internally triggered by switches (keyboard as a matrix of keys) doesn't lose note-off. So it's obviously midi somewhere. I haven't tried the serial midi yet ... (nor build that octo-coupler circuit for it) but I plan to. So my guess is that midi gets lost like a buffer over-run or something? I tried servicing the midi more often and it seemed to improve it but ... its too subjective.

**Reply by marcel-licence** [reply]
Do you have aftertouch active? I've found a little bug which can cause this problem

**Comment by marcel-licence** [comment]
I've made an update. If the adc module is active it will update all controllers. If nothing is connected all values will be set to zero. It's now turned off by default

**Comment by romix123** [comment]
No.
I have been designing some pcb’s to go with your code.
First one is a small board with 8 pots, each equipped with a neopixel, a small oled display 128x32, a pcm5102 dac board, a neopixel, a micro usb port for power and a micro usb port that acts as a USB host port if an otg cable is attached, otherwise it is a usb client. The processor on the board is a SAMD21. That was a mistake. 
Not enough power. 

In a few days, the second rev will be delivered. It has all the above, plus a second pcm5102 board and an ESP32 s2 (esp wroom) board, connected to the samd21 through serial and SPI.

The second board can use the samd as coprocessor for midi in, handling pots, neopixels and display, while the ESP can focus on producing sound.

As a token of appreciation for your work (and soliciting your help to actually get the board doing amazing things), I want to send you a fully populated board if you’re interested. For that I will need a postal address.

All the best, 

Ronald

Verstuurd vanaf mijn iPad

**Reply by marcel-licence** [reply]
Sounds great and really interesting.
I really appreciate that. I will send you an Email.
Moving away from a breadboard circuit does also help a lot and helps also to avoid random problems (like sticking notes using MIDI).
